src/bpk1.rs

```diff
@@ -123,22 +127,83 @@ where
         Self::new_from_bpk1_blocks(blocks)
     }
 
+    fn bytes_from_bpk1_blocks(blocks: Vec<BPK1Block>) -> GenericResult<Vec<u8>> {
+        let mut result = Vec::<u8>::new();
+        let mut writer = Cursor::new(&mut result);
+
+        writer.write(b"BPK1")?;
+        writer.write_u32_le(blocks.len() as u32)?;
+        writer.write_u32_le(7)?;
+
+        let file_size_pos = writer.position();
+        writer.write_u32_le(0)?;
+
+        let header_size_pos = writer.position();
+        writer.write_u32_le(0)?;
+
+        writer.write(&[0; 0x2c])?; // padding
+
+        let header_start_pos = writer.position();
+
+        const BLOCK_HEADER_SIZE: u8 = 0x4 + 0x4 + 0x4 + 0x8;
```

Comment:
Might be better to put this const outside the scope of this function, and write a doc comment explaining where these magic numbers come from